extension/entitystore/serviceprovider.go

```diff
@@ -64,6 +64,7 @@ type serviceprovider struct {
 	done             chan struct{}
 	logger           *zap.Logger
 	mutex            sync.RWMutex
+	logMutex         sync.RWMutex
```

Comment:
With this fix, can we remove `extension/entitystore` from the data race ignore list https://github.com/aws/amazon-cloudwatch-agent/blob/main/Makefile#L206?

Reply:
There's more besides this thatll require a bigger restructuring to address :(